pallets/summary/src/lib.rs

```diff
@@ -318,6 +322,16 @@ pub mod pallet {
     pub type VotingPeriod<T: Config<I>, I: 'static = ()> =
         StorageValue<_, BlockNumberFor<T>, ValueQuery>;
 
+    #[pallet::storage]
+    #[pallet::getter(fn anchor_roots_counter)]
+    pub type AnchorRootsCounter<T: Config<I>, I: 'static = ()> = StorageValue<_, u32, ValueQuery>;
+
+    // Roots created to be anchored to other chains (apart from Ethereum)
+    #[pallet::storage]
+    #[pallet::getter(fn anchor_roots)]
+    pub type AnchorRoots<T: Config<I>, I: 'static = ()> =
+        StorageMap<_, Blake2_128Concat, u32, H256, ValueQuery>;
```

Comment:
why do we need a separate set of storage items? the pallets original storage items should be enough.

Reply:
The previous data structure was not designed to allow easy access for roots after they have been generated. Since the anchoring service needs to do this frequently and db read/write is not cheap and looping through all entries of a map is not sustainable, this new storage item solves these problems.